format-all.el

```diff
@@ -729,7 +735,9 @@ unofficial languages IDs are prefixed with \"_\"."
            (not (null (symbol-value 'flow-minor-mode)))
            "_Flow")
       (and (equal major-mode 'f90-mode) "_Fortran 90")
-      (and (equal major-mode 'ledger-mode) "_Ledger")))
+      (and (equal major-mode 'gleam-mode) "_Gleam")
+      (and (equal major-mode 'ledger-mode) "_Ledger")
+      (language-id-buffer)))
```

Comment:
`(language-id-buffer)` moved to the end because it always returns "C", so anything defined after it was being formatted with `clang-format`.

Reply:
It's fine if it's at the end, but something is definitely wrong if it always returns `"C"`. If you try M-x `eval-expression` `(language-id-buffer)` with different major modes, does it really return `"C"` in e.g. Lisp modes and others?

Reply:
Hm. I thought I saw this in places other than gleam files, but now I cannot reproduce it. I definitely returns "C" when I run it in a gleam file though.

Reply:
The answer lies [in `gleam-mode.el`](https://github.com/gleam-lang/gleam-mode/blob/f3d7838b23f47ce939177edb2eae6104df0780c7/gleam-mode.el#L22): `gleam-mode` is derived from `c-mode`. Format-all uses clang-format for the latter, so in the absence of an explicit `gleam-mode` rule in [`language-id.el`](https://github.com/lassik/emacs-language-id/blob/master/language-id.el) the former inherits it.

More generally, If mode C inherits from mode P, then C must come before P in `language-id-buffer` or `format-all--language-id-buffer`.

Reply:
If you come across other wrongly identified modes, please report them as bugs. The Emacs derived-mode mechanism sometimes causes surprises.